diff -up nx-firewall-master/kcm/package/contents/ui/main.qml.3~ nx-firewall-master/kcm/package/contents/ui/main.qml
--- nx-firewall-master/kcm/package/contents/ui/main.qml.3~	2020-05-12 06:02:39.000000000 +0200
+++ nx-firewall-master/kcm/package/contents/ui/main.qml	2021-01-07 20:22:25.882986564 +0100
@@ -21,21 +21,23 @@
 
 import QtQuick 2.6
 import QtQuick.Layouts 1.3
-import QtQuick.Controls 2.12 as QQC2
+import QtQuick.Controls 1.4
+import QtQuick.Controls.Styles 1.4
+
+import org.kde.plasma.core 2.0 as PlasmaCore
+import org.kde.plasma.components 2.0 as PlasmaComponents
+import org.kde.plasma.extras 2.0 as PlasmaExtras
 
 import org.kde.kcm 1.0
 import org.nomad.ufw 1.0
 import org.nomad.netstat 1.0
 
-import org.kde.kirigami 2.4 as Kirigami
-import org.kde.kcm 1.1 as KCM
-
-KCM.SimpleKCM {
-    id: root
+Item {
+    id: mainWindow
 
-    implicitHeight: Kirigami.Units.gridUnit * 22
-
-    KCM.ConfigModule.quickHelp: i18n("This module lets you configure firewall.")
+    implicitWidth: units.gridUnit * 44
+    implicitHeight: units.gridUnit * 50
+    clip: true
 
     UfwClient {
         id: ufwClient
@@ -51,49 +53,65 @@ KCM.SimpleKCM {
         anchors.fill: parent
     }
 
-    ColumnLayout {
-        QQC2.TabBar {
-            id: tabButtons
-
-            QQC2.TabButton {
-                text: i18n("Rules")
-            }
-            QQC2.TabButton {
-                text: i18n("Connections")
-            }
-            QQC2.TabButton {
-                text: i18n("Logs")
-            }
-        }
-
-        StackLayout {
-            id: tabGroup
-            currentIndex: tabButtons.currentIndex
-
-            Layout.fillWidth: true
-            Layout.fillHeight: true
-
-            RulesView {
-                Layout.fillWidth: true
-                Layout.fillHeight: true
-            }
-
-            ConnectionsView {
-                Layout.fillWidth: true
-                Layout.fillHeight: true
-            }
-
-            LogsView {
-                Layout.fillWidth: true
-                Layout.fillHeight: true
-            }
+    PlasmaCore.FrameSvgItem {
+       anchors.fill: parent
+       imagePath: "dialogs/background"
+       enabledBorders: PlasmaCore.FrameSvg.NoBorder
+    }
+
+    PlasmaComponents.TabBar {
+        id: tabButtons
+        anchors.top: parent.top
+        anchors.left: parent.left
+
+        PlasmaComponents.TabButton {
+            text: i18n("Rules")
+            tab: rulesTab
+        }
+        PlasmaComponents.TabButton {
+            text: i18n("Connections")
+            tab: connectionsTab
+        }
+        PlasmaComponents.TabButton {
+            id: logsTabButton
+            text: i18n("Logs")
+            tab: logsTab
         }
+    }
 
-        QQC2.Label {
-            text: ufwClient.status
+    PlasmaComponents.TabGroup {
+        id: tabGroup
+        anchors.top: tabButtons.bottom
+        anchors.left: parent.left
+        anchors.right: parent.right
+        anchors.bottom: parent.bottom
+        anchors.margins: 12
+
+        PlasmaExtras.ConditionalLoader {
+            id: rulesTab
+            when: tabGroup.currentTab == rulesTab
+            source: Qt.createComponent("RulesView.qml")
+        }
+        PlasmaExtras.ConditionalLoader {
+            id: connectionsTab
+            when: tabGroup.currentTab == connectionsTab
+            source: Qt.createComponent("ConnectionsView.qml")
+        }
+        PlasmaExtras.ConditionalLoader {
+            id: logsTab
+            when: tabGroup.currentTab == logsTab
+            source: Qt.createComponent("LogsView.qml")
         }
     }
 
+    PlasmaComponents.Label {
+        anchors.bottom: parent.bottom
+        anchors.left: parent.left
+        anchors.right: parent.right
+
+        text: ufwClient.status
+    }
+
     function createRuleFromConnection(protocol, localAddress, foreignAddres, status) {
         // Transform to the ufw notation
         localAddress = localAddress.replace("*", "")
